modules/javafx.graphics/src/main/java/com/sun/glass/ui/Window.java

```diff
@@ -285,6 +296,54 @@ protected Window(Window owner, Screen screen, int styleMask) {
         }
     }
 
+    /**
```

Comment:
since the majority of windows is expected to be DECORATED, would it make more sense to move these properties into a separate container akin to `Node.getMiscProperties()` ?

Reply:
Sure, but the downside is that everything gets more bloated for little benefit. Right now, these properties are accessed in derived classes without any checks, since they are final and non-null. In this case, I think ease of use is more important.

Reply:
ease of use for whom?

bloated code: a few bytes in extra methods
bloated data: thousand of bytes added per instance

Since we are adding properties to Window, it's probably ok because the number of instances is relatively low (a real app won't have hundreds of windows).
Still, I would like to recommend investigating the alternative: the memory is not an unlimited resource, and in my experience, it's worth thinking about it.

Reply:
Ease of use for JavaFX developers. Easily maintainable code is one of the most important aspects, miles ahead of chasing bytes and chasing nanoseconds. There's no tangible value in spending any considerable amount of time on thinking how to shave 150 bytes off of a window.

Reply:
a) we are _constantly_ adding runtime bytes, we have to be aware of the cost
b) I think we should rather focus on the application developers, and they do not see the complexity behind the API.  And moving a rarely used stuff into a rarely initialized object should not be a problem for any JavaFX developer, or am I mistaken?

Reply:
Here's the problem: we're also constantly adding code that needs to be written, reviewed, tested, and maintained. We're talking about a small amount of code here in this particular example, but then again, what particular straw broke the camel's back?

I think we have to be aware of runtime overhead where it matters most: in hot loops, or where something is instantiated tens of thousands of times. In all other cases, focusing on bytes and nanoseconds can actually be detrimental in the long run. It makes code inherently harder to reason about, and when many such "optimizations" are taken together, we can end up with code that you can't change any more for fear of breaking the myriads of code paths.